VMR main builds broken: GenerateSdkRuntimeIdentifierChain MSB4216 task host failure since SDK code flow #7064

## Summary

All `main` branch VMR official builds (dnceng/internal, pipeline 1330) have been failing since June 5 with `MSB4216` when the `GenerateSdkRuntimeIdentifierChain` task attempts to launch via `TaskHostFactory`. The failure is on **every platform** (Linux x64/arm64/arm, Windows x64/x86/arm64, macOS arm64).

## Error

```
src/sdk/src/Layout/redist/targets/GenerateLayout.targets(26,5): error MSB4216:
Could not run the "GenerateSdkRuntimeIdentifierChain" task because MSBuild could not
create or connect to a task host with runtime "NET" and architecture "x64".
Please ensure that (1) the requested runtime and/or architecture are available on the
machine, and (2) that the required executable
"/__w/1/s/.dotnet/sdk/11.0.100-preview.5.26227.104/MSBuild" exists and can be run.
```

## Bisection

- **Last passing build**: [2992950](https://dev.azure.com/dnceng/internal/_build/results?buildId=2992950) (commit `025cac314d7`, June 5 ~11:00 UTC)
- **First failing build**: [2993147](https://dev.azure.com/dnceng/internal/_build/results?buildId=2993147) (commit `d233c67796b`, June 5 ~16:00 UTC)

The introducing commit range includes **SDK code flow [dotnet/dotnet#7064](https://github.com/dotnet/dotnet/pull/7064)** (`44865609a40`) as the prime suspect.

## Key findings

### 1. `redist.csproj` was not previously built in VMR

In the passing build (2992950), searching the Linux_x64 log (59,294 lines) for `GenerateSdkRuntimeIdentifierChain`, `RuntimeIdentifierChain`, `TaskHostFactory`, or `redist ->` returns **zero matches**. The `redist.csproj` build target was not being reached.

In the failing build, the log shows:
```
redist -> /__w/1/s/src/sdk/artifacts/bin/redist/Release/net11.0/redist.dll
error MSB4216: ...
```

Something in SDK code flow #7064 changed the build graph so that `redist.csproj` is now actively built in the VMR where it wasn't before.

### 2. `dotnetup` runs in VMR despite `from_vmr` guard

`configure-toolset.sh` was updated to call `InstallBootstrapSdkWithDotnetup` at the top level. It has a guard:
```bash
if [[ "$restore" != true || "$from_vmr" == true || -z "$dotnet_sdk_version" ]]; then
    return
fi
```

However, the VMR build log shows `dotnetup` **is running**:
```
Installing SDK(s) '11.0.100-preview.5.26227.104' to '/__w/1/s/src/sdk/.dotnet' via dotnetup...
Downloading https://aka.ms/dotnet/dotnetup/daily/dotnetup-linux-x64
dotnetup installed successfully to /__w/1/s/src/sdk/eng/dotnetup
```

The VMR does pass `--fromVMR` (`repo-projects/Directory.Build.props:128`), and `build.sh` parses it before sourcing `configure-toolset.sh`, so `from_vmr` should be `true`. Unclear why the guard isn't catching it.

### 3. Task host handshake or launch failure

The `GenerateSdkRuntimeIdentifierChain` task is declared with `TaskFactory="TaskHostFactory" Runtime="NET"` in `sdk-tasks.InTree.targets`. This forces MSBuild to spawn an out-of-process task host using the `MSBuild` apphost at `{sdkPath}/MSBuild`.

The code path in `NodeProviderOutOfProcTaskHost.ResolveAppHostOrFallback()`:
- `MSBuild` apphost exists at `.dotnet/sdk/11.0.100-preview.5.26227.104/MSBuild` → `useAppHost=true`
- `DOTNET_HOST_PATH` is set (`/__w/1/s/.dotnet/dotnet`) → `DOTNET_ROOT` should be derived correctly
- The apphost is launched but the parent/child handshake fails → MSB4216

This may be related to dotnet/msbuild#13879 (task host handshake architecture mismatch).

## Impact

- **All `main` branch builds are broken** — 0% clean pass rate since June 5
- Code flow PRs to `main` (aspnetcore, runtime, roslyn, efcore) are blocked from validation
- No new BAR assets are being published from `main`

## Affected builds

| Build | Date | Commit |
|---|---|---|
| [2994109](https://dev.azure.com/dnceng/internal/_build/results?buildId=2994109) | June 7 | `7150f20` |
| [2993907](https://dev.azure.com/dnceng/internal/_build/results?buildId=2993907) | June 6 | `01a993d` |
| [2993793](https://dev.azure.com/dnceng/internal/_build/results?buildId=2993793) | June 6 | `af0627d` |
| [2993688](https://dev.azure.com/dnceng/internal/_build/results?buildId=2993688) | June 6 | `af0627d` |
| [2993396](https://dev.azure.com/dnceng/internal/_build/results?buildId=2993396) | June 5 | `ba0c241` |
| [2993249](https://dev.azure.com/dnceng/internal/_build/results?buildId=2993249) | June 5 | `3ab08f7` |
| [2993147](https://dev.azure.com/dnceng/internal/_build/results?buildId=2993147) | June 5 | `d233c67` |

## Suggested investigation

1. Why is `redist.csproj` now being built in the VMR when it wasn't before? Check if the SDK code flow changed the traversal/graph build entry points.
2. Why is `InstallBootstrapSdkWithDotnetup` executing despite the `from_vmr` guard?
3. Consider whether `GenerateSdkRuntimeIdentifierChain` needs `TaskHostFactory` at all, or if it can run in-process.

/cc @dotnet/dotnet-cli

--- a/eng/common/SetupNugetSources.ps1
+++ b/eng/common/SetupNugetSources.ps1
@@ -32,6 +32,11 @@ $ErrorActionPreference = "Stop"
 Set-StrictMode -Version 2.0
 [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
 
+# This script only consumes helper functions from tools.ps1 to configure NuGet feeds.
+# Skip importing configure-toolset.ps1 so that repo-specific toolset setup (e.g. acquiring
+# a bootstrap SDK) is not triggered as a side effect of feed configuration.
+$disableConfigureToolsetImport = $true
+
 . $PSScriptRoot\tools.ps1
 
 # Adds or enables the package source with the given name

--- a/eng/common/SetupNugetSources.sh
+++ b/eng/common/SetupNugetSources.sh
@@ -40,6 +40,11 @@ while [[ -h "$source" ]]; do
 done
 scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
 
+# This script only consumes helper functions from tools.sh to configure NuGet feeds.
+# Skip importing configure-toolset.sh so that repo-specific toolset setup (e.g. acquiring
+# a bootstrap SDK) is not triggered as a side effect of feed configuration.
+disable_configure_toolset_import=1
+
 . "$scriptroot/tools.sh"
 
 if [ ! -f "$ConfigFile" ]; then

